infoschema/tables.go

```diff
@@ -1172,10 +1158,21 @@ func dataForTables(ctx sessionctx.Context, schemas []*model.DBInfo) ([][]types.D
 				if table.GetPartitionInfo() != nil {
 					createOptions = "partitioned"
 				}
-				autoIncID, err := getAutoIncrementID(ctx, schema, table)
-				if err != nil {
-					return nil, err
+				var autoIncID interface{}
+				hasAutoIncID := false
+				for _, col := range table.Cols() {
+					if mysql.HasAutoIncrementFlag(col.Flag) {
+						hasAutoIncID = true
```

Comment:
Could we use `hasAutoIncrementColumn` in ddl_api.go?

Reply:
done.